packages/authentication/src/types.ts

```diff
@@ -92,14 +116,18 @@ export const USER_PROFILE_NOT_FOUND = 'USER_PROFILE_NOT_FOUND';
 /**
  * Registers an authentication strategy as an extension of the
  * AuthenticationBindings.AUTHENTICATION_STRATEGY_EXTENSION_POINT_NAME extension
- * point.
+ * point. Initiates regististration of security spec if provided.
+ *
  * @param context - Context object
  * @param strategyClass - Class for the authentication strategy
  */
 export function registerAuthenticationStrategy(
   context: Context,
   strategyClass: Constructor<AuthenticationStrategy>,
 ) {
+  const spec = strategyClass.prototype.constructor.securitySpec;
```

Comment:
Please note that `strategyClass.prototype` has type any, therefore everything after that is not type-checked by the compiler. Let's find a way how to preserve compiler checks for `securitySpec`.

First of all, I think we should access `securitySpec` directly via the class constructor:

```suggestion
  const spec = strategyClass.securitySpec;
```

Now this is obviously rejected by the compiler because `Constructor<>` does not have any `securitySpec` property.

```
ts(2339)
Property 'securitySpec' does not exist on type 'Constructor<AuthenticationStrategy>' 
```

A possible solution is to change the type of `strategyClass` to describe `securitySpec` argument.

```diff
-   strategyClass: Constructor<AuthenticationStrategy>,
+   strategyClass: Constructor<AuthenticationStrategy> & {securitySpec?: () => SecurityScheme; }
```

(This addresses the "static property" part of my previous comment.)

To make the "strategy class" type easier to use in other places, I suggest to create a type alias:

```ts
export type AuthenticationStrategyClass = Constructor<AuthenticationStrategy> & {
  securitySpec?: () => SecurityScheme; 
}
```

Reply:
I had no idea you could simply append an argument description.  Something new learned, thank you! 🙇